code/modules/supply/supply_order.dm

```diff
@@ -0,0 +1,101 @@
+/datum/supply_order
+	///The orders ID number
+	var/ordernum
+	///The supply pack this order is for
+	var/datum/supply_packs/object
+	///The person who ordered the supply order
+	var/orderedby
+	///The occupation/assignment of the person who ordered the supply order
+	var/orderedbyRank
+	///The account tied to this order, important for personal orders
+	var/orderedbyaccount
+	///The station department datum of the person's department who ordered the supply order
+	var/datum/station_department/ordered_by_department
+	///Reason/purpose for order given by orderer
+	var/comment
+	///amount of crates purchased
+	var/crates
+	///does this order need to be approve by the department head?
+	var/requires_head_approval = FALSE
+	///does this order need to be approve by the QM?
+	var/requires_qm_approval = FALSE
+
+/obj/item/paper/request_form
+	name = "request form"
+	var/order_number
+
+/datum/supply_order/proc/createObject(atom/_loc, errors = 0)
+	if(!object)
+		return
+
+	//create the crate
+	var/atom/Crate = new object.containertype(_loc)
+	Crate.name = "[object.containername] [comment ? "([comment])":"" ]"
+	if(object.access)
+		Crate:req_access = list(text2num(object.access))
```

Comment:
No common type here you can use?

Reply:
im terrified of touching any of this code

Reply:
fixed